### PR TITLE
[WIP] aws route53 recovery readiness new resource - cells

### LINKED ISSUE #20384: Add support for Cell resource from Route53 Recovery Readiness
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Adds support for the Cell resource from Route53 Recovery Readiness.  Cells are used to monitor fault boundaries in AWS Route53 Recovery Readiness.

### New or Affected Resource(s)

* aws_resource_route53_recovery_readiness_cell

### Potential Terraform Configuration

N\A

### References

* Depends on #20367 (Addition of new service)
* https://docs.aws.amazon.com/recovery-readiness/latest/api/create-cell.html

### PR DESCRIPTION
**Pull request by WyatyPeert**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #20384

### WIP Questions
1. Route53 Recovery Readiness Cells expose their ARNs in responses under the attribute "cell_arn".  Should the schema implement just `arn`, just `cell_arn` or both. Related to [Implements Arn Attributes](https://github.com/hashicorp/terraform-provider-aws/blob/main/docs/contributing/pullrequest-submission-and-lifecycle.md#changelog-process)
2. In the NestedCell test, the ChildCell only gets a value for `parent_readiness_scopes` after the ParentCell gets updated. I had to put an additional step in so that the ChildCell would be re-read and actually return have a `parent_readiness_scopes` value.  Is there a better way to do this?

### WIP Notes
Note that the service has not yet been merged so the diff on this CR includes the service in addition to the first resource, though I do not intend to merge them that way.

Output from acceptance testing:

```
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAwsRoute53RecoveryReadinessCell_ -timeout 180m
=== RUN   TestAccAwsRoute53RecoveryReadinessCell_basic
=== PAUSE TestAccAwsRoute53RecoveryReadinessCell_basic
=== RUN   TestAccAwsRoute53RecoveryReadinessCell_nestedCell
=== PAUSE TestAccAwsRoute53RecoveryReadinessCell_nestedCell
=== RUN   TestAccAwsRoute53RecoveryReadinessCell_tags
=== PAUSE TestAccAwsRoute53RecoveryReadinessCell_tags
=== CONT  TestAccAwsRoute53RecoveryReadinessCell_basic
=== CONT  TestAccAwsRoute53RecoveryReadinessCell_nestedCell
=== CONT  TestAccAwsRoute53RecoveryReadinessCell_tags
--- PASS: TestAccAwsRoute53RecoveryReadinessCell_basic (22.44s)
--- PASS: TestAccAwsRoute53RecoveryReadinessCell_tags (50.27s)
--- PASS: TestAccAwsRoute53RecoveryReadinessCell_nestedCell (51.93s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	54.771s
...
```

### References
* https://docs.aws.amazon.com/recovery-readiness/latest/api/create-cell.html

### CODE REVIEW

File: aws/resource_aws_route53_recovery_readiness_cell_test.go [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+					resource.TestCheckResourceAttr(resourceNameParent, "parent_readiness_scopes.#", "0"),
+					testAccCheckAwsRoute53RecoveryReadinessCellExists(resourceNameChild),
+					testAccMatchResourceAttrGlobalARN(resourceNameChild, "cell_arn", "route53-recovery-readiness", regexp.MustCompile(`cell/.+`)),
+					resource.TestCheckResourceAttr(resourceNameChild, "cells.#", "0"),
+				),
+			},
+			{
+				Config: testAccAwsRoute53RecoveryReadinessCellParentConfig(rNameChild, rNameParent),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceNameChild, "parent_readiness_scopes.#", "1"),
```
Comment (AdamTylerLynch) [inline aws/resource_aws_route53_recovery_readiness_cell_test.go:76]: @WyatyPeert is parent_readiness_scopes now resolving > 0? If not, let’s see if the community has any ideas for how to make that attribute refresh.

Comment (WyatyPeert) [reply aws/resource_aws_route53_recovery_readiness_cell_test.go:76]: @AdamTylerLynch Yup! As currently written, the scopes resolve to 1 like I'd expect. It just seems a touch misleading to have an additional test step w/ Config solely to get the attribute refreshed.

File: aws/provider.go [hunk 2]
```diff
@@ -995,6 +995,7 @@ func Provider() *schema.Provider {
 			"aws_route53_resolver_query_log_config_association":       resourceAwsRoute53ResolverQueryLogConfigAssociation(),
 			"aws_route53_resolver_rule_association":                   resourceAwsRoute53ResolverRuleAssociation(),
 			"aws_route53_resolver_rule":                               resourceAwsRoute53ResolverRule(),
+			"aws_route53_recovery_readiness_cell":                     resourceAwsRoute53RecoveryReadinessCell(),
```
Comment (ewbankkit) [inline aws/provider.go:998]: See comments in #20367.

```suggestion
			"aws_route53recoveryreadiness_cell":                     resourceAwsRoute53RecoveryReadinessCell(),
```

Comment (WyatyPeert) [reply aws/provider.go:998]: Will go through and make these changes then rev.

**Pull request closed without merging**